specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/WebApps.json

```diff
@@ -18918,6 +20271,65 @@
         }
       }
     },
+    "FunctionStatus": {
+      "description": "Function status.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "FunctionStatus resource specific properties",
+          "properties": {
+            "errors": {
+              "description": "Collection of initialization errors for the function.",
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "HostKeys": {
+      "description": "Functions host level keys.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
```

Comment:
The return body doesn't contain any of the `ProxyOnlyResource` properties. I have an open PR created as part of an attempt to get the function key listed in Terraform where I saw the response body for calls didn't contain an `id`, `name` etc property.

https://github.com/Azure/azure-rest-api-specs/issues/7143

Response Body:
```json
{
 "masterKey":"Gfbg2Tv4lwZW/a8fLZzLr7sgQYmHau9rQWERl4yG4P34REDACTED",
 "functionKeys":{"default":"Sr4Kth4Io86N1OOTEXhkJeqq99bN2TWLXREDACTED=="},
 "systemKeys":{}
}
```


Reference: Proxy Only Resource from `commonDefinitions.json`

```
    "ProxyOnlyResource": {
      "description": "Azure proxy only resource. This resource is not tracked by Azure Resource Manager.",
      "properties": {
        "id": {
          "description": "Resource Id.",
          "type": "string",
          "readOnly": true
        },
        "name": {
          "description": "Resource Name.",
          "type": "string",
          "readOnly": true
        },
        "kind": {
          "description": "Kind of resource.",
          "type": "string"
        },
        "type": {
          "description": "Resource type.",
          "type": "string",
          "readOnly": true
        }
      },
      "x-ms-azure-resource": true
    },
```

Reply:
All the spec changes in this PR are generated by our automated build process from annotations on the response types declared for our APIs. I've followed the existing pattern in our codebase for all of these, and this is how ALL the response types for our App Service APIs are handled (as you can see by looking at other non-Functions APIs in this spec). So if there are any issues, they aren't specific to the changes I'm making here, and would have to be addressed separately across the board for all our APIs.  @naveedaz knows more about how this generation works and can hopefully shed some light.

Reply:
Hi @mathewc

> So if there are any issues, they aren't specific to the changes I'm making here, and would have to be addressed separately across the board for all our APIs.

I believe the issue is specific to the changes you are making here. This comment is on spec changed in this PR. In the changes, the additional APIs added do not function as this REST spec defines. What is the purpose of the spec if it doesn't accurately describe the behaviour of the API?

Or, have I missed something here that would reconcile this issue?

I agree that fixing  other occurrences of this issue, on APIs not updated in this PR, should be addressed as a separate step.

Reply:
fixed